terraform/files/bin/fixup_flavor_volumes.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# fixup_flavor_volumes.sh
+#
+# Usage: fixup_flavor_volumes.sh CCCFG CLUSTERTEMPLATE
+#  Check if the flavor has a disk that's large enough
+#  Change machine template to allocate root disk of sufficient size if no disk exists
+#
+# (c) Kurt Garloff <scs@garloff.de>, 6/2023
+# SPDX-License-Identifier: Apache-2.0
+
+usage()
+{
+	echo "Usage: fixup_flavor_volumes.sh CLUSTERCTL CLUSTERTEMPLATE"
+	exit 1
+}
+
+if test -z "$2"; then usage; fi
+
+UBU_IMG_NM=$(yq eval '.OPENSTACK_IMAGE_NAME' $1)
+CTRLFLAVOR=$(yq eval '.OPENSTACK_CONTROL_PLANE_MACHINE_FLAVOR' $1)
+WORKFLAVOR=$(yq eval '.OPENSTACK_NODE_MACHINE_FLAVOR' $1)
+
+flavor_disk.sh "$CTRLFLAVOR" "$UBU_IMG_NM"
+DISKCTRL=$?
+flavor_disk.sh "$WORKFLAVOR" "$UBU_IMG_NM"
+DISKWORK=$?
+
+if test $DISKCTRL -ge 128; then echo "ERROR $((DISKCTRL-256)) using ctrlplane flavor $CTRLFLAVOR for image $UBU_IMG_NM"; exit 1; fi
+if test $DISKWORK -ge 128; then echo "ERROR $((DISKWORK-256)) using worker flavor $WORKFLAVOR for image $UBU_IMG_NM"; exit 1; fi
```

Comment:
Two reasons:
(a) I would not want node images that require root volumes > 100GiB
(b) The way the wanted size is returned from the helper uses the exit code where >= 128 indicates an error. The exit code is a single byte ...

We could of course use a different mechanism (e.g. echo the needed size to stdout and read it `$(helper)`), but this way it could be kept a bit simpler. If you see the need for larger volumes, please explain, so we can change this.